src/feat/wave-reader.cc

```diff
@@ -227,49 +227,73 @@ void WaveData::Read(std::istream &is) {
     // format of RIFF files that means that the input may sometimes be padded
     // with 1 unused byte to make the total size even.
-    KALDI_ERR << "Expected " << riff_chunk_size << " bytes in RIFF chunk, but "
-              << "after first data block there will be " << riff_chunk_read
-              << " + " << data_chunk_size << " bytes "
-              << "(we do not support reading multiple data chunks).";
-  }
-
-  std::vector<char*> data_pointer_vec;
-  std::vector<int> data_size_vec;
-  uint32 num_bytes_read = 0;
-  for (int32 remain_chunk_size = data_chunk_size; remain_chunk_size > 0;
-       remain_chunk_size -= kBlockSize) {
-    int32 this_block_size = remain_chunk_size;
-    if (kBlockSize < remain_chunk_size)
-      this_block_size = kBlockSize;
-    char *block_data_vec = new char[this_block_size];
-    is.read(block_data_vec, this_block_size);
-    num_bytes_read += is.gcount();
-    data_size_vec.push_back(is.gcount());
-    data_pointer_vec.push_back(block_data_vec);
-    if (num_bytes_read < this_block_size)
-      break;
-  }
-
-  std::vector<char> chunk_data_vec(num_bytes_read);
-  uint32 data_address = 0;
-  for (int i = 0; i < data_pointer_vec.size(); i++) {
-    memcpy(&(chunk_data_vec[data_address]), data_pointer_vec[i],
-           data_size_vec[i]);
-    delete[] data_pointer_vec[i];
-    data_address += data_size_vec[i];
+    if ((riff_chunk_size == 0 && data_chunk_size == 0) 
+         || (riff_chunk_size == 4294967295 && data_chunk_size == 4294967295))
+        KALDI_WARN << "Read in " << riff_chunk_size << " bytes in RIFF chunk, "
+                   << "Read in " << riff_chunk_read << " bytes in data chunk."
+                   << "maybe data is read in stream mode"
+                   << "so that we don't know exact size in advance";
+    else
+        KALDI_ERR << "Expected " << riff_chunk_size << " bytes in RIFF chunk, but "
+                  << "after first data block there will be " << riff_chunk_read
+                  << " + " << data_chunk_size << " bytes "
+                  << "(we do not support reading multiple data chunks).";
   }
 
+  std::vector<char> chunk_data_vec;
   char *data_ptr = &(chunk_data_vec[0]);
-  if (num_bytes_read == 0 && num_bytes_read != data_chunk_size) {
-    KALDI_ERR << "WaveData: failed to read data chunk (read no bytes)";
-  } else if (num_bytes_read != data_chunk_size) {
-    KALDI_ASSERT(num_bytes_read < data_chunk_size);
-    KALDI_WARN << "Read fewer bytes than specified in the header: "
-               << num_bytes_read << " < " << data_chunk_size;
+  uint32 num_bytes_read = 0;
+  if (!((riff_chunk_size == 0 && data_chunk_size == 0)
+     || (riff_chunk_size == 4294967295 && data_chunk_size == 4294967295))) {
+     std::vector<char*> data_pointer_vec;
+     std::vector<int> data_size_vec;
+     num_bytes_read = 0;
+     for (int32 remain_chunk_size = data_chunk_size; remain_chunk_size > 0;
```

Comment:
Sorry, I can't quite understand which lines you comment on. The data bytes are now accumulated using std::vector<char> chunk_data_vec; (based on std::vector<char*> data_pointer_vec; and std::vector<int> data_size_vec;) when is_stream_mode == false